packages/core/src/components/tabs2/tabs2.tsx

```diff
@@ -150,9 +150,15 @@ export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
         }
     }
 
-    public componentDidUpdate(_prevProps: ITabs2Props, prevState: ITabs2State) {
+    public componentDidUpdate(prevProps: ITabs2Props, prevState: ITabs2State) {
         if (this.state.selectedTabId !== prevState.selectedTabId) {
             this.moveSelectionIndicator();
+        } else if (prevState.selectedTabId != null) {
+            const prevChildren = JSON.stringify(prevProps.children.map(thisArg => JSON.stringify(thisArg.props, ["title", "id"])));
+            const children = JSON.stringify(React.Children.map(this.props.children, thisArg => JSON.stringify(thisArg.props, ["title", "id"])));
```

Comment:
hmmmm this is clever but i'm not comfortable relying on this sort of logic in __core__: i've seen issues where certain JSX elements are actually _not_ stringifiable (as in, `JSON.stringify` throws an error).

what we really want is a `shallowCompare`, but we unfortunately don't have such a utility in __core__ yet (we do in __table__).

Reply:
ok, and are there any plans to add `shallowCompare` into **core**?

Reply:
@cmslewis if trivial enough, would you be able to add a `shallowCompare` util you're using in `table` into `core`? Looks like we need it to move forward with this PR

Reply:
@franciscofabian @llorca just moved the utils in question into **core** in https://github.com/palantir/blueprint/pull/1604. @franciscofabian, does this unblock you for this PR?

Reply:
@cmslewis yes, it does! I will update this PR to use the new utils